### ISSUE TITLE
seatunnel同步postgresql中json类型数据报错

### ISSUE THREAD

**Issue by wulinzhai** [apache/seatunnel#5103]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

seatunnel 同步postgresql中json类型数据报错

### SeaTunnel Version

2.3.2

### SeaTunnel Config

```conf
env {
  execution.parallelism = 1
  job.mode = "BATCH"
}

source {
    Jdbc {
        result_table_name = "source-table"
        url = "jdbc:postgresql://xxxxx:5432/xxxxx"
        driver = "org.postgresql.Driver"
        connection_check_timeout_sec = 100
        user = "xxxxx"
        password = "xxxxxxxx"
        query = "select id, name, type, data from public.scene limit 16"
    }
}

transform {
  FieldMapper {
    source_table_name = "source-table"
    result_table_name = "transform-table"
    field_mapper = {
        id = id
        type = type
        name = name
        data = data
    }
  }
}

sink {
  Console {
    source_table_name = "transform-table"
  }
}
```

### Running Command

```shell
../bin/seatunnel.sh --config pg.template -e local
```

### Error Exception

```log
2023-07-18 02:57:03,816 ERROR org.apache.seatunnel.core.starter.SeaTunnel - Reason:SeaTunnel job executed failed 

2023-07-18 02:57:03,818 ERROR org.apache.seatunnel.core.starter.SeaTunnel - Exception StackTrace:org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:188)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34)
Caused by: org.apache.seatunnel.api.common.PrepareFailException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: java.lang.IllegalArgumentException: The number of field names must be the same as the number of field types.
        at org.apache.seatunnel.connectors.seatunnel.jdbc.source.JdbcSource.prepare(JdbcSource.java:108)
        at org.apache.seatunnel.engine.core.parse.JobConfigParser.parseSource(JobConfigParser.java:81)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSource(MultipleTableJobConfigParser.java:298)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:159)
        at org.apache.seatunnel.engine.client.job.JobExecutionEnvironment.getLogicalDag(JobExecutionEnvironment.java:155)
        at org.apache.seatunnel.engine.client.job.JobExecutionEnvironment.execute(JobExecutionEnvironment.java:147)
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:140)
        ... 2 more
 
2023-07-18 02:57:03,820 ERROR org.apache.seatunnel.core.starter.SeaTunnel - 
===============================================================================



Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:188)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34)
Caused by: org.apache.seatunnel.api.common.PrepareFailException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: java.lang.IllegalArgumentException: The number of field names must be the same as the number of field types.
        at org.apache.seatunnel.connectors.seatunnel.jdbc.source.JdbcSource.prepare(JdbcSource.java:108)
        at org.apache.seatunnel.engine.core.parse.JobConfigParser.parseSource(JobConfigParser.java:81)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSource(MultipleTableJobConfigParser.java:298)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:159)
        at org.apache.seatunnel.engine.client.job.JobExecutionEnvironment.getLogicalDag(JobExecutionEnvironment.java:155)
        at org.apache.seatunnel.engine.client.job.JobExecutionEnvironment.execute(JobExecutionEnvironment.java:147)
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:140)
        ... 2 more
```

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wulinzhai** [comment]
CREATE TABLE "public"."scene" (
  "id" int8 NOT NULL DEFAULT nextval('scene_id_seq'::regclass),
  "name" text COLLATE "pg_catalog"."default" NOT NULL,
  "type" text COLLATE "pg_catalog"."default" NOT NULL,
  "data" jsonb NOT NULL
)
;

pg表中data是json格式的数据

**Comment by Lifu12** [comment]
please view https://seatunnel.apache.org/docs/2.3.2/concept/schema-feature support type

**Comment by jackyyyyyssss** [comment]
assign me

**Comment by jackyyyyyssss** [comment]
@wulinzhai fix 2.3.3
